indexer/services/comlink/src/lib/helpers.ts

```diff
@@ -531,3 +534,36 @@ export function checkIfValidDydxAddress(address: string): boolean {
   const pattern: RegExp = /^dydx[0-9a-z]{39}$/;
   return pattern.test(address);
 }
+
+/* ------- PNL HELPERS ------- */
+
+/**
+ * Aggregates a list of PnL ticks, combining any PnL ticks for the same blockheight by summing
+ * the equity, totalPnl, and net transfers.
+ * Returns a map of block height to the resulting PnL tick.
+ * @param pnlTicks 
+ * @returns 
+ */
+export function aggregatePnlTicks(
+  pnlTicks: PnlTicksFromDatabase[],
+): Map<number, PnlTicksFromDatabase> {
+  const aggregatedPnlTicks: Map<number, PnlTicksFromDatabase> = new Map();
+  for (const pnlTick of pnlTicks) {
+    const blockHeight: number = parseInt(pnlTick.blockHeight, 10);
+    if (aggregatedPnlTicks.has(blockHeight)) {
+      const currentPnlTick: PnlTicksFromDatabase = aggregatedPnlTicks.get(
+        blockHeight,
+      ) as PnlTicksFromDatabase;
+      aggregatedPnlTicks.set(blockHeight, {
+        ...currentPnlTick,
+        equity: (parseFloat(currentPnlTick.equity) + parseFloat(pnlTick.equity)).toString(),
+        totalPnl: (parseFloat(currentPnlTick.totalPnl) + parseFloat(pnlTick.totalPnl)).toString(),
+        netTransfers: (parseFloat(currentPnlTick.netTransfers) +
+            parseFloat(pnlTick.netTransfers)).toString(),
+      });
```

Comment:
in extreme scenarios, these summations might overflow?

Reply:
The largest float possible is in the order of 1e308, so it's impossible given that these are human read-able dollar amounts.
The largest float possible without loss of precision (with 2 decimal places) I think is ~1e16, which is still near impossible with human-readable dollar amounts.